test/Sentry.Unity.Tests/ScreenshotAttachmentContentTests.cs

```diff
@@ -0,0 +1,93 @@
+using System.IO;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace Sentry.Unity.Tests
+{
+    public class ScreenshotAttachmentTests : DisabledSelfInitializationTests
+    {
+        private class Fixture
+        {
+            public SentryUnityOptions Options = new() { AttachScreenshot = true };
+            public bool IsMainThread = true;
+
+            public ScreenshotAttachmentContent GetSut()
+            {
+                var gameObject = new GameObject("TestSentryMonoBehaviour");
+                var sentryMonoBehaviour = gameObject.AddComponent<SentryMonoBehaviour>();
+                sentryMonoBehaviour.IsMainThread = () => IsMainThread;
+
+                return new ScreenshotAttachmentContent(Options, sentryMonoBehaviour);
+            }
+        }
+
+        private Fixture _fixture = null!;
+
+        [SetUp]
+        public new void Setup() => _fixture = new Fixture();
+
+        [Test]
+        [TestCase(ScreenshotQuality.High, 1920)]
+        [TestCase(ScreenshotQuality.Medium, 1280)]
+        [TestCase(ScreenshotQuality.Low, 854)]
+        public void GetTargetResolution_ReturnsTargetMaxSize(ScreenshotQuality quality, int expectedValue)
+        {
+            var actualValue = ScreenshotAttachmentContent.GetTargetResolution(quality);
+
+            Assert.AreEqual(expectedValue, actualValue);
+        }
+
+        [Test]
+        public void GetStream_IsMainThread_ReturnsStream()
+        {
+            _fixture.IsMainThread = true;
+            var sut = _fixture.GetSut();
+
+            var stream = sut.GetStream();
+
+            Assert.IsNotNull(stream);
+        }
+
+        [Test]
+        public void GetStream_IsNonMainThread_ReturnsNullStream()
+        {
+            _fixture.IsMainThread = false;
```

Comment:
And just in case you didn't want to do the above changes & fixture - you could just run this test in a different thread?

Reply:
Wow, that was a real case of tunnel vision right there.